pep-9999.rst

```diff
@@ -0,0 +1,988 @@
+``sysconfig.get_default_scheme()``, but based on
+configuration (e.g.  ``pip install --user``), it may use a
+different scheme.
+
+Whenever the installer is installing to a ``sysconfig``
+scheme, this PEP specifies that the installer should never
+modify or delete files outside of that scheme. For instance,
+if it's upgrading a package, and the package is already
+installed in a directory outside that scheme (perhaps in a
+directory from another scheme), it should leave the existing
+files alone.
+
+If the installer does end up shadowing an existing
+installation during an upgrade, we recommend that it
+produces a warning at the end of its run.
+
+If the installer is installing to a location outside of a
+``sysconfig`` scheme (e.g., ``pip install --target``), then
+this subsection does not apply.
+
+Recommendations for distributions
+=================================
+
+This section is non-normative. It provides advice we think
+is a good idea for distributions to follow unless they have
+a specific reason otherwise.
+
+* Distributions should create an ``EXTERNALLY-MANAGED`` file
+  in their ``stdlib`` directory.
+
+* Distributions that produce official images for
+  single-application containers (e.g., Docker container
+  images) should remove the ``EXTERNALLY-MANAGED`` file,
+  preferably in a way that makes it not come back if a user
+  of that image installs package updates inside their image
+  (think ``RUN apt-get dist-upgrade``). On dpkg-based
+  systems, using ``dpkg-divert --local`` to persistently
+  rename the file would work. On other systems, there may
+  need to be some configuration flag available to a
+  post-install script to re-remove the
+  ``EXTERNALLY-MANAGED`` file.
+
+* The file should contain a useful and distribution-relevant
+  error message indicating both how to install system-wide
+  packages via the distribution's package manager and how to
+  set up a virtual environment. If your distribution is
+  often used by users in a state where the ``python3``
+  command is available (and especially where ``pip`` or
+  ``get-pip`` is available) but ``python3 -m venv`` does not
+  work, the message should indicate clearly how to make
+  ``python3 -m venv`` work properly.
+
+* Consider packaging pipx_, a tool for installing
+  Python-language applications, and suggesting it in the
+  error. It automatically creates a virtual environment for
+  that application alone, which is a much better default for
+  end users who want to install some Python-language
+  software (which isn't available in the distribution) but
+  are not themselves Python users. Packaging pipx in the
+  distribution avoids the irony of instructing users to
+  ``pip install --user --break-system-packages pipx`` to
+  _avoid_ breaking system packages.  Consider arranging
+  things so your distribution's package / environment for
+  Python when used directly (e.g., ``python3`` on Fedora or
+  ``python3-full`` on Debian) depends on pipx in addition to
+  pip.
```

Comment:
`python3-full` doesn't depend on `pip`. It does depend on `ensurepip`, so that `venv` works, but you only get `pip` inside the virtualenv, not outside it.